### PR TITLE
Add documentation for memberAPI

### PR DESCRIPTION
**Pull request by oscarwang20**
### Summary 

- See title
- Changed `body` parameters to `member`
- removed unused `request` parameter in `updateMember`

### CODE REVIEW

**Review by kevinmram — APPROVED** [review]
LGTM!

File: backend/src/API/memberAPI.ts [hunk 1]
```diff
@@ -7,56 +6,86 @@ import { getNetIDFromEmail, computeMembersDiff } from '../utils/memberUtil';
...
 export const allApprovedMembers = (): Promise<readonly IdolMember[]> =>
   MembersDao.getAllMembers(true);
 
-export const setMember = async (body: IdolMember, user: IdolMember): Promise<IdolMember> => {
+/**
+ * Creates a new IDOL member given their credentials.
+ * @param member - the new `IdolMember` to create.
+ * @param user - the `IdolMember` making the request.
+ * @throws `PermissionError` if `user` does not have permission to edit members.
+ * @throws `badRequestError` if the email of the new `IdolMember` is undefined.
```
Comment (andrew032011) [inline backend/src/API/memberAPI.ts:27]: nit `BadRequestError`

File: backend/src/API/memberAPI.ts [hunk 2]
```diff
@@ -7,56 +6,86 @@ import { getNetIDFromEmail, computeMembersDiff } from '../utils/memberUtil';
...
+ * @returns the newly created `IdolMember`.
+ */
+export const setMember = async (member: IdolMember, user: IdolMember): Promise<IdolMember> => {
   const canEdit = await PermissionsManager.canEditMembers(user);
   if (!canEdit) {
     throw new PermissionError(
       `User with email: ${user.email} does not have permission to edit members!`
     );
   }
-  if (!body.email || body.email === '') {
```
Comment (andrew032011) [inline backend/src/API/memberAPI.ts:22]: 👍

File: backend/src/api.ts [hunk 3]
```diff
@@ -218,7 +218,7 @@ loginCheckedDelete('/member/:email', async (req, user) => {
   return {};
 });
 loginCheckedPut('/member', async (req, user) => ({
-  member: await updateMember(req, req.body, user)
```
Comment (andrew032011) [inline backend/src/api.ts:221]: 👍

**Review by andrew032011 — APPROVED** [review]

**Pull request merged**